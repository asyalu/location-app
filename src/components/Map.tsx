import { FC, useCallback, useRef } from 'react';
import { GoogleMap, useJsApiLoader } from '@react-google-maps/api';

interface IMap {
  center: {
    lat: number;
    lng: number;
  };
}

const containerStyle = {
  width: '400px',
  height: '400px',
};

const defaultOptions = {
  streetViewControl: false,
  fullscreenControl: false,
};

const Map: FC<IMap> = ({ center }): JSX.Element => {
  const { isLoaded } = useJsApiLoader({
    id: 'google-map-script',
    googleMapsApiKey: process.env.REACT_APP_MAPS_KEY!,
  });

  const mapRef = useRef<google.maps.Map | undefined>(undefined);

  const onLoad = useCallback((map: google.maps.Map) => {
    mapRef.current = map;
  }, []);

  const onUnmount = useCallback(() => {
    mapRef.current = undefined;
  }, []);

  return isLoaded ? (
    <GoogleMap
      mapContainerStyle={containerStyle}
      center={center}
      zoom={10}
      onLoad={onLoad}
      onUnmount={onUnmount}
      options={defaultOptions}
    >
      <>1</>
    </GoogleMap>
  ) : (
    <span>Loading...</span>
  );
};

export { Map };
