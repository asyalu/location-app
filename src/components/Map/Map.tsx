import { FC } from 'react';
import { GoogleMap, useJsApiLoader, Marker } from '@react-google-maps/api';
import { IMap } from '../../interfaces/map';

const containerStyle = {
  width: '300px',
  height: '200px',
};

const defaultOptions = {
  streetViewControl: false,
  fullscreenControl: false,
};

const Map: FC<IMap> = ({ lat, lng }): JSX.Element => {
  const { isLoaded } = useJsApiLoader({
    id: 'google-map-script',
    googleMapsApiKey: process.env.REACT_APP_MAPS_KEY as string,
  });

  return isLoaded ? (
    <GoogleMap
      mapContainerStyle={containerStyle}
      center={{ lat, lng }}
      zoom={10}
      options={defaultOptions}
    >
      <Marker position={{ lat, lng }} />
    </GoogleMap>
  ) : (
    <span>Loading...</span>
  );
};

export { Map };
