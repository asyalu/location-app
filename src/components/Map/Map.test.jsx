import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Map } from './Map';

test('renders map with provided coordinates', () => {
  const mockCoordinates = { lat: 40.7128, lng: -74.006 };
  const { asFragment } = render(
    <Map
      lat={mockCoordinates.lat}
      lng={mockCoordinates.lng}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
