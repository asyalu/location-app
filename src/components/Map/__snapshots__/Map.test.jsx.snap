// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders map with provided coordinates 1`] = `
<DocumentFragment>
  <span>
    Loading...
  </span>
</DocumentFragment>
`;
